app/controllers/scim_rails/application_controller.rb

```diff
@@ -81,6 +101,90 @@ def process_path(operation)
         acc
       end
     end
+
+    # `process_filter_path` is a method that, like the `process_path`
+    # method above, parses the string in the "path" key of a PATCH
+    # operation.  It converts the operation into a Hash that resembles
+    # the User or Group schema, and it will be easier to fetch
+    # attributes needed to update a user or group.
+    #
+    # Example: given the following operation:
+    #   operation = {
+    #     'op': 'Add',
+    #     'path': 'emails[type eq \'work\'].value',
+    #     'value': 'Wolfe'
+    #   }
+    # then calling `process_filter_path(operation)` will return the
+    # Hash:
+    #   {
+    #     emails: [
+    #       {
+    #         type: 'work'
+    #         value: 'Wolfe'
+    #       }
+    #     ]
+    #   }
+    # which will will later be processed to fetch the attributes to be
+    # updated by the PATCH request
+    def process_filter_path(operation)
+      path_string = operation["path"]
+
+      pre_bracket_path = before_square_brackets(path_string)
+      filter = inside_square_brackets(path_string)
+
+      args = filter.split(' ')
+
+      key = args[0]
+      value = inside_quotations(args[2])
+
+      inner_hash = {}
+      inner_hash[key] = value
+      inner_hash["value"] = operation["value"]
+
+      full_hash = {}
+      full_hash[pre_bracket_path] = [ inner_hash ]
+
+      full_hash
+    end
+
+    # `multi_attr_type_to_value` is a method that takes a request body
+    # and compares it to the `scim_attribute_type_mappings`, to return
+    # a hash which may contain attributes of the User model
+    #
+    # Example: given the following:
+    #   schema_hash = {
+    #     "emails": [
+    #       {
+    #         "type": "work",
+    #         "value": "work@example.com"
+    #       }
+    #     ]
+    #   }
+    # and `scim_attribute_type_mappings` being defined as:
+    #   {
+    #     "emails" => {
+    #       "work" => :email
+    #     }
+    #   }
+    # means calling `multi_attr_type_to_value` will return the Hash
+    #   {
+    #     email: "work@example.com"
+    #   }
+    def multi_attr_type_to_value(schema_hash)
+      merge_params = {}
+
+      ScimRails.config.scim_attribute_type_mappings.each do |mapping_key, mapping_value|
+        if schema_hash.key?(mapping_key)
+          schema_hash[mapping_key].each do |attribute|
+            if mapping_value.key?(attribute["type"])
+              merge_params[mapping_value[attribute["type"]]] = attribute["value"]
+            end
+          end
+        end
+      end
+
+      merge_params
```

Comment:
🌵 getting a little deeply nested here, try to avoid that in general

You could do a couple things to reduce that nesting.
- You could literally use [`#reduce`](https://apidock.com/ruby/Enumerable/reduce) and remove the setup and return lines
- **You could change line 177 to be a guard clause, which would remove one level of nesting**
- Line 179 could be made in to a guard with `next` instead of `return`
- In the event you couldn't do any of those things, you could move the contents of the `#each` out to its own method